lisp/ess-inf.el

```diff
@@ -1293,43 +1293,44 @@ wrapping the code into:
         ;; Swap the process buffer with the output buffer before
         ;; sending the command
         (unwind-protect
-            (progn
-              (when use-sentinel
-                (process-put proc 'ess-output-sentinel sentinel))
-              (set-process-buffer proc out-buffer)
-              (set-process-filter proc 'inferior-ess-ordinary-filter)
-              (with-current-buffer out-buffer
-                (ess-setq-vars-local proc-alist)
-                (setq buffer-read-only nil)
-                (erase-buffer)
-                (set-marker (process-mark proc) (point-min))
-                (inferior-ess-mark-as-busy proc)
-                (process-send-string proc rich-cmd)
-                ;; Need time for ess-create-object-name-db on PC
-                (if no-prompt-check
-                    (sleep-for 0.02)   ; 0.1 is noticeable!
-                  (unless (ess-wait-for-process proc nil wait force-redisplay timeout)
-                    (error "Timeout during background ESS command `%s'"
-                           (ess--strip-final-newlines cmd)))
-                  ;; Remove prompt. If output is cat(..)ed without a
-                  ;; final newline, this deletes the last line of output.
+            (unwind-protect
+                (progn
+                  (when use-sentinel
+                    (process-put proc 'ess-output-sentinel sentinel))
+                  (set-process-buffer proc out-buffer)
+                  (set-process-filter proc 'inferior-ess-ordinary-filter)
+                  (with-current-buffer out-buffer
+                    (ess-setq-vars-local proc-alist)
+                    (setq buffer-read-only nil)
+                    (erase-buffer)
+                    (set-marker (process-mark proc) (point-min))
+                    (inferior-ess-mark-as-busy proc)
+                    (process-send-string proc rich-cmd)
+                    ;; Need time for ess-create-object-name-db on PC
+                    (if no-prompt-check
+                        (sleep-for 0.02) ; 0.1 is noticeable!
+                      (unless (ess-wait-for-process proc nil wait force-redisplay timeout)
+                        (error "Timeout during background ESS command `%s'"
```

Comment:
I think it's a good idea to increase the timeout but to me it makes sense to throw when the R process doesn't respond. If we can't interrupt when requested it doesn't seem safe to continue normal execution, the calling context of an `ess-command` is likely to assume it can interact with the process.

Reply:
I am not saying we shouldn't interrupt. I am saying we can avoid throwing. We are erroring there just to catch it two line latter by ourselves. The only reason to do that is to send the message to the user.

Reply:
oops I thought we were discussing about the timeout error inside `ess-interrupt` but you're suggesting we don't timeout-error in `ess-command` altogether?

I strongly think a timeout in `ess-command` should abort everything. We had a lot of hanging issues lately (most of them now solved) and having a clean error improved ux _a lot_.

> The only reason to do that is to send the message to the user.

It is also to interrupt the caller of `ess-command`. If the command failed, that code should be aborted. Why should it continue if the command could not be run to completion?

Also in the future we'll bubble R errors to emacs errors here.

Reply:
> just to catch it two line latter by ourselves

We don't catch the error by the way, we let it go through.

Reply:
Arh, right. I missed that part. I guess I am getting elisp-rusty a bit :/